src/trace.ts

```diff
@@ -16,16 +16,17 @@
 
 'use strict';
 
-/**
- * Creates a trace object.
- * @constructor
- */
-function Trace(projectId, traceId) {
-  this.projectId = projectId;
-  this.traceId = traceId;
-  this.spans = [];
-}
+import { TraceSpan } from './trace-span';
 
-module.exports = Trace;
+export class Trace {
+  public readonly spans: TraceSpan[] = [];
```

Comment:
Do you have any style thoughts on declaring fields in multiple places (inside constructor + at the top of the class)?

Reply:
It's one or the other, declaring in both places is a TS error.

Placing in the constructor is a style guide preference: https://g3doc.corp.google.com/javascript/typescript/g3doc/styleguide.md?cl=head#class-members

Reply:
Sorry, I meant declaring some of a classes fields at the top and others in the constructor. It seems like it would be nice to have one place to look to see what state is maintained by a class.

Reply:
I agree that it's nicer to have it all in one place; the style guide is not specific on whether fields should be split. Maybe we can re-organize all classes as a followup.